mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+            Ok(s) => s,
+            Err(_) => "failed to execute vgs".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    let r: VGsReport = match serde_json::from_slice(output.stdout.as_slice()){
+        Ok(r) => r,
+        Err(e) => return Err(Error::FailedParsing{err: e.to_string()}),
+    };
+
+    let capacity: u64 = match r.report[0].vg[0].vg_size.parse(){
+        Ok(c) => c,
+        Err(e) => return Err(Error::FailedParsing{err: e.to_string()})
+    };
+    let free: u64 = match r.report[0].vg[0].vg_free.parse(){
+        Ok(c) => c,
+        Err(e) => return Err(Error::FailedParsing{err: e.to_string()})
+    };
+
+    let output = Command::new(PVS_COMMAND)
+        .args(&["--options=vg_name,pv_name", "--noheadings", "--reportformat=json"])
+        .output()
+        .await
+       .expect("failed to execute vgs");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute pvs".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    let r: PVsReport = match serde_json::from_slice(output.stdout.as_slice()){
+        Ok(r) => r,
+        Err(e) => return Err(Error::FailedParsing{err: e.to_string()}),
+    };
+
+    let mut disks: Vec<String> = vec![];
+
+    for p in r.report[0].pv.as_slice() {
+        if p.vg_name == name{
+            disks.push(p.pv_name.as_str().to_string())
+        }
+    }
+
+    let pool = Pool{
+        name: name.to_string(),
+        disks,
+        capacity,
+        used: capacity - free,
+        pooltype: PoolType::Lvm.into(),
+        state: PoolState::PoolOnline.into(),
+    };
+    Ok(pool)
+}
+
+pub async fn remove_lvm_vg(name: String) -> Result<(), Error> {
+    let pool = match get_lvm_vg(name.clone()).await{
+        Ok(p) => p,
+        Err(e) => return Err(e),
+    };
+
+    let output = Command::new(VGREMOVE_COMMAND)
+        .arg(name.as_str())
+        .output()
+        .await
+        .expect("failed to execute vg_remove");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vg_remove".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    let output = Command::new(PVREMOVE_COMMAND)
+        .args(pool.disks.as_slice())
+        .output()
+        .await
+       .expect("failed to execute pv_remove");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute pv_remove".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+    info!("pool '{}' has been destroyed successfully", name);
+    Ok(())
+}
+
+pub async fn lookup_lvm_disk(name: &str) -> Option<String> {
```

Comment:
it seems this function is only used to determine if the disk in use, as such -- would a pool not suffice? if not, and we need to return more context, why not return the while struct Pool?

Reply:
The list_vol_group gets the pools which have `mayastor` tag on them. This function list all the vol groups in the system and check if the disk provided in `lvs` pool creation is not in use by a vol group

Reply:
I understand, but it is not what I tried asking... Why do we return String? Why not Option<Pool>? Also should we not use udev to obtain this information? It would be a lot nice to report it that way maybe. We can improve the udev usage later.